src/query/string_opfunc.c

```diff
@@ -4580,33 +4490,30 @@ db_string_rlike (const DB_VALUE * src_string, const DB_VALUE * pattern, const DB
 	}
     }
 
-  /* match against pattern; regexec returns zero on match */
-  rx_err = cub_regexec (rx_compiled_regex, src_char_string_p, src_length, 0, NULL, 0);
-  switch (rx_err)
+  // *INDENT-OFF*
+  try
     {
-    case CUB_REG_OKAY:
-      *result = V_TRUE;
-      break;
-
-    case CUB_REG_NOMATCH:
-      *result = V_FALSE;
-      break;
-
-    default:
-      rx_err_len = (int) cub_regerror (rx_err, rx_compiled_regex, rx_err_buf, REGEX_MAX_ERROR_MSG_SIZE);
+      std::string src (src_char_string_p, src_length);
+      bool match = std::regex_search (src, *rx_compiled_regex);
+      *result = match ? V_TRUE : V_FALSE;
+    }
+  catch (std::regex_error & e)
+    {
+      // regex execution exception, error_complexity or error_stack
       error_status = ER_REGEX_EXEC_ERROR;
-      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error_status, 1, rx_err_buf);
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error_status, 1, e.what ());
       *result = V_ERROR;
-      break;
+      goto cleanup;
```

Comment:
"goto cleanup" might not be needed because the code would be reached to "cleanup".